tests/integration-tests/tests/schedulers/test_custom_munge_key.py

```diff
@@ -0,0 +1,194 @@
+       can be submitted.
+    2. Rotation prep: Attempt munge key rotation without stopping compute and login nodes, expecting error messages.
+    3. Compute fleet shutdown: Prepare for munge key rotation scenario.
+    4. Key rotation without login node stop: Attempt rotation, checking for expected failure and error messages.
+    5. Login node stoppage and munge key rotation: Update cluster to stop login nodes, then execute munge key rotation.
+    6. Munge key removal: Update cluster to remove the custom munge key. Test Munge Key has been changed
+    7. Roll back with failure: Update cluster to add back the custom munge key. But let the cluster update fail after
+       the custom munge key has been added. Trigger cluster roll back. Test if munge key is fully functional.
+    """
+    encode_custom_munge_key = create_base64_encoded_key()
+    custom_munge_key_arn = store_secret_in_secret_manager(
+        region,
+        secret_string=encode_custom_munge_key,
+    )
+    cluster_config = pcluster_config_reader(custom_munge_key_arn=custom_munge_key_arn)
+    cluster = clusters_factory(cluster_config, upper_case_cluster_name=True)
+
+    # Test if the munge key was successfully fetched, decoded and shared in HeadNode and LoginNodes
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    _test_custom_munge_key_fetch_and_decode(remote_command_executor, encode_custom_munge_key)
+    _test_munge_key_shared(remote_command_executor)
+
+    remote_command_executor_login = RemoteCommandExecutor(cluster, use_login_node=True)
+    _test_custom_munge_key_fetch_and_decode(remote_command_executor_login, encode_custom_munge_key)
+    remote_command_executor_login.close_connection()
+
+    # Test if compute node can run jobs, indicating the munge key was successfully fetched.
+    scheduler_commands = scheduler_commands_factory(remote_command_executor)
+    scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "command": "srun sleep 1",
+            "nodes": 2,
+        }
+    )
+
+    # Test error message when both compute and login nodes are not stopped.
+    _test_update_munge_key_without_stop_login_or_compute(remote_command_executor)
+
+    # Stop compute fleets
+    cluster.stop()
+    wait_for_computefleet_changed(cluster, "STOPPED")
+
+    # Test error message when login nodes are not stopped.
+    _test_update_munge_key_without_stop_login_or_compute(remote_command_executor, compute_stopped=True)
+
+    # Update cluster with pcluster.stop_login.config.yaml to stop login nodes.
+    update_cluster_stop_login_config = pcluster_config_reader(
+        config_file="pcluster.stop_login.config.yaml",
+        custom_munge_key_arn=custom_munge_key_arn,
+    )
+    cluster.update(str(update_cluster_stop_login_config))
+
+    # wait for LoginNodes gracetime_period
+    check_login_nodes_stopped(remote_command_executor)
```

Comment:
Nitpick: please add an underscore in the name of this function so that it's _clear_ that this should only be used by this test (because this function would not work without custom munge key).

Reply:
Done!